### PR TITLE
Extend Probe Helper for use with CloudPubSubSource probe

### PR DESCRIPTION
**Pull request by tommyreddad**
Fixes #

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/test_images/probe_helper/probe_helper.go [hunk 1]
```diff
@@ -41,16 +51,28 @@ type envConfig struct {
...
 		log.Printf("Received probe request: %+v \n", event)
 		eventID := event.ID()
 		receivedEvents[eventID] = make(chan bool, 1)
 		ctx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Minute)
 		defer cancel()
-		if res := c.Send(ctx, event); !cloudevents.IsACK(res) {
-			log.Printf("Error when sending event %v to broker: %+v \n", eventID, res)
-			return res
+		switch event.Type() {
+		case "broker-e2e-delivery-probe", "broker-e2e-delivery-probe-kubectl-exec", "broker-e2e-delivery-probe-kubectl-exec-warm-up":
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:62]: We can change the event type to just "broker-e2e-delivery-probe" and "cloudpubsubsource-probe". We could use other attributes such as Ce-Source to further differentiate probes of the same event type if needed

File: test/test_images/probe_helper/probe_helper.go [hunk 2]
```diff
@@ -63,10 +88,40 @@ func forwardFromProbe(ctx context.Context, c cloudevents.Client, receivedEvents
...
+}
+
+type message struct {
+	Attributes cloudEvent `json:"attributes"`
+}
+
+type eventData struct {
+	Message message `json:"message"`
+}
+
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:102]: are there similar types in the pubsub package?

File: test/test_images/probe_helper/probe_helper.yaml [hunk 3]
```diff
@@ -33,11 +33,23 @@ spec:
         - name: probe-helper
           image: ko://github.com/google/knative-gcp/test/test_images/probe_helper
           env:
+            - name: GOOGLE_APPLICATION_CREDENTIALS
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.yaml:36]: as a follow up, we will modify the setup script to configure the clusters with this secret

File: test/test_images/probe_helper/probe_helper.go [hunk 4]
```diff
@@ -63,10 +88,40 @@ func forwardFromProbe(ctx context.Context, c cloudevents.Client, receivedEvents
...
+			// pubsub sender client, and encoded as data in a CloudEvent by the CloudPubSubSource.
+			data := eventData{}
+			err := json.Unmarshal(event.Data(), &data)
+			if err != nil {
+				log.Printf("Failed to unmarshal event data: %v", err)
+				return cloudevents.ResultACK
+			}
+			eventID = data.Message.Attributes.ID
+		default:
+			log.Printf("Unrecognized event type")
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:122]:
```suggestion
			log.Printf("Unrecognized event type: %v", event.Type())
```

File: test/test_images/probe_helper/probe_helper.go (hunk 1 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go]: Done.

File: test/test_images/probe_helper/probe_helper.go (hunk 2 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go]: Yes, for the Pub/Sub Message. I couldn't find out a clean way to read the inner CloudEvent from the Pub/Sub Attributes map through, so I just access the "ce-id" key now.

File: test/test_images/probe_helper/probe_helper.go [hunk 5]
```diff
@@ -107,11 +171,10 @@ func runProbeHelper() {
...
 	// make a map to store the channel for each event
 	receivedEvents := make(map[string]chan bool)
-	ctx := signals.NewContext()
-	// start a goroutine to receive the event from probe and forward the event to the broker
+	// start a goroutine to receive the event from probe and forward it appropriately
 	log.Println("Starting Probe Helper server...")
-	go sc.StartReceiver(ctx, forwardFromProbe(ctx, sc, receivedEvents, timeout))
-	// Receive the event from the trigger and return the result back to the probe
+	go sc.StartReceiver(ctx, forwardFromProbe(ctx, sc, psc, receivedEvents, timeout))
+	// Receive the event from a trigger and return the result back to the probe
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:177]:
```suggestion
	// Receive the event and return the result back to the probe
```

File: test/test_images/probe_helper/probe_helper.go [hunk 6]
```diff
@@ -66,7 +91,28 @@ func forwardFromProbe(ctx context.Context, c cloudevents.Client, receivedEvents
 func receiveFromTrigger(receivedEvents map[string]chan bool) cloudEventsFunc {
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:91]:
```suggestion
func receiveEvent(receivedEvents map[string]chan bool) cloudEventsFunc {
```

**Pull request merged**